### ISSUE TITLE
Touch devices dragging problem

### ISSUE THREAD

**Issue by marvinIsSacul** [oakmac/chessboardjs#160]
Hi Mr Chris Oakman,

I have noticed that when chessboard.js is viewed and used on a touch screen device, piece dragging becomes a problem - it is quite annoying because as you'd try to drag a certain piece, the entire view gets dragged and scrolls. And as a result, the piece you want to move doesn't end up at another board rank/file.

As a work-around I did the following via jQuery

<code><pre>
jQuery('#chess_board').on('scroll touchmove touchend touchstart contextmenu', function(e){
       e.preventDefault();
   });
</pre></code>

**Comment by jonsubs** [comment]
Hi marvinIsSacul,
Where did you use the statement above? I'm pretty new to JS and JQuery and do not really know where should I place your code.
Thanks, Jon.

**Comment by marvinIsSacul** [comment]
After initializing the chess board.

**Comment by psoleckimoj** [comment]
Any updates on this bug? I've seen a couple of PRs for fixing it but no-one has looked at them and both now have conflicts against Master. We're moving to devices that have touch and are running into this issue so would be nice to fixed. Plus more and more devices are touch-enabled so would be a sensible feature to have.

**Comment by oakmac** [comment]
I agree chessboard.js needs better support for touch devices. It has been in the back of my mind to update; just has not been a priority lately ;)

**Comment by dngo** [comment]
> Hi Mr Chris Oakman,
> 
> I have noticed that when chessboard.js is viewed and used on a touch screen device, piece dragging becomes a problem - it is quite annoying because as you'd try to drag a certain piece, the entire view gets dragged and scrolls. And as a result, the piece you want to move doesn't end up at another board rank/file.
> 
> As a work-around I did the following via jQuery
> 
> 
> ```
> 
> jQuery('#chess_board').on('scroll touchmove touchend touchstart contextmenu', function(e){
> e.preventDefault();
> });
> ```



> Hi Mr Chris Oakman,
> 
> I have noticed that when chessboard.js is viewed and used on a touch screen device, piece dragging becomes a problem - it is quite annoying because as you'd try to drag a certain piece, the entire view gets dragged and scrolls. And as a result, the piece you want to move doesn't end up at another board rank/file.
> 
> As a work-around I did the following via jQuery
> 
> 
> ```
> 
> jQuery('#chess_board').on('scroll touchmove touchend touchstart contextmenu', function(e){
> e.preventDefault();
> });
> ```

Thanks a lot, this solved my issue with my website chesshub.com

**Comment by dngo** [comment]
@marvinIsSacul  thanks a lot, this solve the issue I was having.

**Comment by s-mv** [comment]
A temporary way of making it work is using on touchstart, make the piece have absolutely position, move it at touch position every touch move and finally, on touchend, run a for loop checking every square and then see if the piece overlaps any, and if more that one, which one the most. Whew!

**Comment by devdesj** [comment]
Is there a way to avoid having to drag pieces and instead just clicking on the squares?

**Comment by leonbrag** [comment]
it seems still an issue...

**Comment by theduffster** [comment]
For anyone like me who was going crazy trying to both stop touch events from scrolling the board, but also get that initial touch event so you can play audio upon player piece placement in IOS Safari, hopefully this is helpful:

```
//Initialize chess board
board = Chessboard('chessBoard', config)

//Create audio instance
const dropSound = new Audio();
dropSound.autoplay = true;

// Plays blank sound once to initialize audio for IOS
function blankSound(){
    myFunc = function(){}; // kill it as soon as it was called so it only plays once
    dropSound.src = "data:audio/wav;base64,UklGRigAAABXQVZFZm10IBIAAAABAAEARKwAAIhYAQACABAAAABkYXRhAgAAAAEA";
};

//Catch touch events and prevent scrolling board when playing
jQuery('#chessBoard').on('scroll touchmove touchend touchstart contextmenu', function(e){
  blankSound();
  e.preventDefault();
});
```

It's impossible to get that initial user interaction to initialize audio with `touch-action: none;`, so need to use this
